docs/adrs/003-Retry_endpoint_design.md

```diff
@@ -35,26 +35,26 @@ The endpoint takes a job or list of job ids and then attempts to resubmit them t
 * EE2 looks up the job versions and parameters, and then submits the job to be retried, incrementing the `retry_count`
   of the job being retried, and the newly launched job gains a pointer to the `_PARENT_RETRY_JOB_ID`
 * The job is submitted and upon successful submission, notifies the `retry_parent` and notifies the `parent_job_id` that a new `child_job` has been added
+* Jobs submitted by an administrator with a ResourceRequirements use resource requirements from the catalog / ee2 config exclusively
 
-
-### Batch Behavior
-* If a job has the attribute of `batch_job=True` the retry will fail, since there is no method to re-run. This is a bug, as it doesn't fail gracefully.
-* If a job has the attribute of `batch_job=True`, but is actually a child job, the parent will be notified of this new retried job
-* Multiple in-flight retries are allowed.
+### Batch Jobs Behavior
+* Adds child_job_id to parent_job_id.child_job_ids
 
 ## Retry_job behavior
-* Blocking and single submit to HTCondor. It should be fine
+* Blocking and single submit to HTCondor. It should be fine as it returns relatively quickly
   
 ## Retry_jobs behavior
-* Submitting multiple jobs uses the `run_job` endpoint, and is blocking (NOT OK!)
+* Submitting multiple jobs uses the `run_job` endpoint, and is blocking. This can cause issues if the network drops, and makes the narrative not aware of the state of the retry. Submitting 100 jobs currently takes 12 seconds, and that is a lot of time for things to go wrong. 
+* Multiple in-flight retries are allowed.
 
 ### Desired Behavior
-* Prevent multiple in-flight retries to prevent the user from wasting their own resources (and the queues resources)
+* Prevent multiple in-flight retries of the same original job to prevent the user from wasting their own resources (and the queues resources)
 * Add retry_count to retried jobs as well to aid in more book-keeping in a new field called `retry_number`
 * Non blocking job submission for submitting multiple jobs, possibly via using `run_job_batch` (requires refactor of run_job_batch)
 * One single submission to HTCondor instead of multiple job submissions
-* Ability to gracefully handle jobs with children
-* Ability to handle database consistentcy during retry failure
+* Ability to handle database consistency during retry failure. (see thread https://github.com/kbase/execution_engine2/pull/383#discussion_r638341940)
+* Add failure conditions in the run() method and see if any of those should be checked prior to starting jobs
+* Prevent retry if resolving `retry_parent` results in two or more of the same job_ids in a retry_jobs request (see thread https://github.com/kbase/execution_engine2/pull/383#discussion_r640907736)
```

Comment:
Or silently just run the one retry job and report the results to both positions in the list

Reply:
Added

Reply:
:+1: